x/payment/keeper/msg_server_withdraw.go

```diff
@@ -4,15 +4,51 @@ import (
 	"context"
 
 	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
 
 	"github.com/bnb-chain/greenfield/x/payment/types"
 )
 
 func (k msgServer) Withdraw(goCtx context.Context, msg *types.MsgWithdraw) (*types.MsgWithdrawResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	// check stream record
+	creator := sdk.MustAccAddressFromHex(msg.Creator)
 	from := sdk.MustAccAddressFromHex(msg.From)
+
+	params := k.GetParams(ctx)
+	if ctx.IsUpgraded(upgradetypes.Nagqu) {
+		if msg.Amount.GTE(params.WithdrawTimeLockThreshold) {
```

Comment:
there is one concern: if we change the `WithdrawTimeLockThreshold` param, the old records are below the new threshold. so we will not check the existing record. maybe we should not check compare the amount here.

Reply:
refactor the code, to handle the case.